dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -2530,122 +2526,128 @@ Spdp::SpdpTransport::handle_input(ACE_HANDLE h)
+        const size_t read = start - buff_.length();
+        if (read < static_cast<size_t>(submessageLength + SMHDR_SZ)) {
+          if (!ser.skip(static_cast<CORBA::UShort>(submessageLength + SMHDR_SZ
+                                                   - read))) {
+            ACE_ERROR((LM_ERROR,
+                       ACE_TEXT("(%P|%t) ERROR: Spdp::SpdpTransport::handle_input() - ")
+                       ACE_TEXT("failed to skip sub message length\n")));
+            return 0;
+          }
         }
-      } else {
-        plist.length(1);
-        const RepoId guid = make_id(header.guidPrefix, ENTITYID_PARTICIPANT);
-        plist[0].guid(guid);
-        plist[0]._d(PID_PARTICIPANT_GUID);
+      } else if (!submessageLength) {
+        break; // submessageLength of 0 indicates the last submessage
       }
-
-      outer_->data_received(data, plist, remote);
-      break;
     }
-    default:
-      SubmessageHeader smHeader;
-      if (!(ser >> smHeader)) {
-        ACE_ERROR((
-              LM_ERROR,
-              ACE_TEXT("(%P|%t) ERROR: Spdp::SpdpTransport::handle_input() - ")
-              ACE_TEXT("failed to deserialize SubmessageHeader for SPDP\n")));
-        return 0;
-      }
-      submessageLength = smHeader.submessageLength;
-      break;
-    }
-    if (submessageLength && buff_.length()) {
-      const size_t read = start - buff_.length();
-      if (read < static_cast<size_t>(submessageLength + SMHDR_SZ)) {
-        if (!ser.skip(static_cast<CORBA::UShort>(submessageLength + SMHDR_SZ
-                                                 - read))) {
-          ACE_ERROR((LM_ERROR,
-            ACE_TEXT("(%P|%t) ERROR: Spdp::SpdpTransport::handle_input() - ")
-            ACE_TEXT("failed to skip sub message length\n")));
-          return 0;
-        }
+  }
+
+  // Handle some RTI protocol multicast to the same address
+  if ((buff_.size() >= 4) && (ACE_OS::memcmp(buff_.rd_ptr(), "RTPX", 4) == 0)) {
+    return 0; // Ignore
+  }
+
+  // Assume STUN
+
+# ifndef ACE_RECVPKTINFO
+  ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: Spdp::SpdpTransport::handle_input() ")
+             ACE_TEXT("potential STUN message received but this version of the ACE ")
+             ACE_TEXT("library doesn't support the local_address extension in ")
+             ACE_TEXT("ACE_SOCK_Dgram::recv\n")));
+  ACE_NOTSUP_RETURN(0);
+# else
+
+#ifdef OPENDDS_SECURITY
+  DCPS::Serializer serializer(&buff_, DCPS::Serializer::SWAP_BE);
+  STUN::Message message;
+  message.block = &buff_;
+  if (serializer >> message) {
+    if (relay_srsm_.is_response(message)) {
+      process_relay_sra(relay_srsm_.receive(message));
+    } else {
+      ICE::Endpoint* endpoint = get_ice_endpoint();
+      if (endpoint) {
+        ICE::Agent::instance()->receive(endpoint, local, remote, message);
       }
-    } else if (!submessageLength) {
-      break; // submessageLength of 0 indicates the last submessage
     }
   }
+#endif
+# endif
```

Comment:
Indent preprocessor lines consistently (or we could just give up and left-align all of them)